tests/gpu-tests/dual_source_blending.rs

```diff
@@ -0,0 +1,221 @@
+            src_factor: wgpu::BlendFactor::Src1Alpha,
+            dst_factor: wgpu::BlendFactor::OneMinusSrc1Alpha,
+            operation: wgpu::BlendOperation::Subtract,
+        },
+    }
+}
+
+#[gpu_test]
+static DUAL_SOURCE_BLENDING_FEATURE_DISABLED: GpuTestConfiguration = GpuTestConfiguration::new()
+    .parameters(TestParameters::default())
+    .run_async(dual_source_blending_disabled);
+
+async fn dual_source_blending_disabled(ctx: TestingContext) {
+    let vertex_shader = ctx.device.create_shader_module(ShaderModuleDescriptor {
+        label: Some("vertex_shader"),
+        source: ShaderSource::Wgsl(VERTEX_SHADER.into()),
+    });
+    let fragment_shader = ctx.device.create_shader_module(ShaderModuleDescriptor {
+        label: Some("fragment_shader"),
+        source: ShaderSource::Wgsl(FRAGMENT_SHADER_WITHOUT_DUAL_SOURCE_BLENDING.into()),
+    });
+
+    // Can't create a render pipeline using blend modes that require dual source blending.
+    fail(
+        &ctx.device,
+        || {
+            let _ = ctx
+                .device
+                .create_render_pipeline(&wgpu::RenderPipelineDescriptor {
+                    label: Some("render_pipeline"),
+                    layout: None,
+                    fragment: Some(wgpu::FragmentState {
+                        module: &fragment_shader,
+                        entry_point: Some("fs_main"),
+                        targets: &[Some(wgpu::ColorTargetState {
+                            format: wgpu::TextureFormat::Rgba8Unorm,
+                            blend: Some(blend_state_with_dual_source_blending()),
+                            write_mask: wgpu::ColorWrites::all(),
+                        })],
+                        compilation_options: Default::default(),
+                    }),
+                    vertex: wgpu::VertexState {
+                        module: &vertex_shader,
+                        entry_point: None,
+                        buffers: &[],
+                        compilation_options: Default::default(),
+                    },
+                    primitive: wgpu::PrimitiveState::default(),
+                    depth_stencil: None,
+                    multisample: wgpu::MultisampleState::default(),
+                    multiview: None,
+                    cache: None,
+                });
+        },
+        Some("Features Features { features_wgpu: FeaturesWGPU(0x0), features_webgpu: FeaturesWebGPU(DUAL_SOURCE_BLENDING) } are required but not enabled on the device"),
+    );
+
+    // Can't create a shader using dual source blending.
+    fail(
+        &ctx.device,
+        || {
+            let _ = ctx.device.create_shader_module(ShaderModuleDescriptor {
+                label: Some("shader"),
+                source: ShaderSource::Wgsl(FRAGMENT_SHADER_WITH_DUAL_SOURCE_BLENDING.into()),
+            });
+        },
+        Some("Capability Capabilities(DUAL_SOURCE_BLENDING) is not supported"),
```

Comment:
thought: "is not supported" is sorta not the truth if the device _could_ have had it enabled. I wonder if it'd be worth filing an issue for fixing the error up in the case that it could have been enabled?

Reply:
Decided to let that one slide and not open more small diagnostics issues ;-)